### PR TITLE
[CLI-3117,CLI-2827] use map[string]string for various outputs

### PR DESCRIPTION
**Pull request by sgagniere**
Release Notes
-------------

Breaking Changes
- The serialized field "subtask_statuses" of `confluent kafka broker task list` is now a map
- The serialized field "config" of `confluent schema-registry exporter describe` is now a map
- The serialized field "kms_properties" of `confluent schema-registry kek` commands is now a map
- The field "Network Zonal Subdomains" of `confluent stream-share consumer redeem` and `confluent stream-share consumer share describe` is now a map
- The serialized field "network_zones" of `confluent stream-share consumer redeem` and `confluent stream-share consumer share describe` is now an array

New Features
- PLACEHOLDER

Bug Fixes
- PLACEHOLDER

Checklist
---------
- [ ] Leave this box unchecked if features are not yet available in production

What
----

References
----------

Test & Review
-------------

### CODE REVIEW

File: pkg/output/table.go [hunk 1]
```diff
@@ -247,3 +247,12 @@ func (t *Table) isMap() bool {
 	_, ok := t.objects[0].(map[string]string)
 	return ok
 }
+
+func isZero(value reflect.Value) bool {
+	// special case: value.IsZero() doesn't evaluate to true for maps that are empty but not nil
```
Comment (brianstrauch) [inline pkg/output/table.go:252]: What about slices that are empty but not `nil`?

File: test/fixtures/output/schema-registry/kek/create.golden [hunk 2]
```diff
@@ -2,7 +2,7 @@
 | Name           | kek-name                            |
 | KMS Type       | AWS_KMS                             |
 | KMS Key ID     | arn:aws:kms:us-west-2:9979:key/abcd |
-| KMS Properties | KeyState="Enabled"                  |
+| KMS Properties | KeyState=Enabled                    |
```
Comment (brianstrauch) [inline test/fixtures/output/schema-registry/kek/create.golden:5]: Is it ok if we make this change @rayokota?

File: test/fixtures/output/schema-registry/exporter/describe.golden [hunk 3]
```diff
@@ -4,6 +4,6 @@
 | Subject Format | my-${subject} |
 | Context Type   | CUSTOM        |
 | Context        | mycontext     |
-| Config         | key1="value1" |
-|                | key2="value2" |
+| Config         | key1=value1   |
```
Comment (brianstrauch) [inline test/fixtures/output/schema-registry/exporter/describe.golden:7]: Let's `git blame` and ask the person who wrote this if this is ok?

File: pkg/output/table.go (hunk 1 above)
Comment (sgagniere) [reply pkg/output/table.go:252]: It probably has the same issue, but I didn't investigate

File: test/fixtures/output/schema-registry/exporter/describe.golden (hunk 3 above)
Comment (sgagniere) [reply test/fixtures/output/schema-registry/exporter/describe.golden:7]: @Clarence97 I think you added the schema exporter commands initially. Is this change ok?

Comment (Clarence97) [reply test/fixtures/output/schema-registry/exporter/describe.golden:7]: Yes, this should be fine

File: pkg/output/table.go (hunk 1 above)
Comment (sgagniere) [reply pkg/output/table.go]: It did have the same problem. I updated the code to handle both.

File: internal/stream-share/command_consumer_share_describe.go [hunk 4]
```diff
@@ -82,8 +82,8 @@ func (c *command) describeConsumerShare(cmd *cobra.Command, args []string) error
 				InviteExpiresAt:          share.GetInviteExpiresAt(),
 				NetworkDnsDomain:         network.GetDnsDomain(),
 				// TODO: Serialize array instead of string in next major version
```
Comment (brianstrauch) [inline internal/stream-share/command_consumer_share_describe.go:84]:
```suggestion
```

**Review by brianstrauch — APPROVED** [review]

**Pull request merged**